nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/parameter/StandardParameterReferenceManager.java

```diff
@@ -128,6 +128,14 @@ private boolean isComponentReferencing(final ComponentNode componentNode, final
             }
         }
 
+        if (componentNode instanceof ProcessorNode) {
```

Comment:
We shouldn't be treating ProcessorNode specially. Perhaps it makes more sense to get rid of this `isComponentReferencing` method from `StandardParameterReferenceManager` and instead add a `isReferencingParameter(String parameterName)` to `ComponentNode`. Then the existing implementation would move to `AbstractComponentNode` and `StandardProcessorNode` can then override the method, call the `super.isReferencingParameter` and if false check its `configurationParameterReferences`